Tests/SuperLinq.Test/HasDuplicatesTest.cs

```diff
@@ -0,0 +1,230 @@
+﻿namespace Test;
+
+public class HasDuplicatesTest
+{
+	[Fact]
+	public void When_Asking_For_Duplicates_On_Sequence_Without_Duplicates_Then_False_Is_Returned()
+	{
+		var stringArray = new[]
+		{
+			"FirstElement",
+			"SecondElement",
+			"ThirdElement"
+		};
+
+		var hasDuplicates = stringArray.HasDuplicates();
+
+		Assert.False(hasDuplicates);
+	}
+
+	[Fact]
+	public void When_Asking_For_Duplicates_On_Sequence_Projection_Without_Duplicates_Then_False_Is_Returned()
+	{
+		var dummyClasses = new DummyClass[]
+		{
+			new("FirstElement"),
+			new("SecondElement"),
+			new("ThirdElement")
+		};
+
+		var hasDuplicates = dummyClasses.HasDuplicates(x => x.ComparableString);
+
+		Assert.False(hasDuplicates);
+	}
+
+	[Fact]
+	public void When_Asking_For_Duplicates_On_Sequence_With_Duplicates_Then_True_Is_Returned()
+	{
+		var stringArray = new[]
+		{
+			"FirstElement",
+			"DUPLICATED_STRING",
+			"DUPLICATED_STRING",
+			"ThirdElement"
+		};
+
+		var hasDuplicates = stringArray.HasDuplicates();
+
+		Assert.True(hasDuplicates);
+	}
+
+	[Fact]
+	public void When_Asking_For_Duplicates_On_Sequence_Projection_With_Duplicates_Then_True_Is_Returned()
+	{
+		var dummyClasses = new DummyClass[]
+		{
+			new("FirstElement"),
+			new("DUPLICATED_STRING"),
+			new("DUPLICATED_STRING"),
+			new("ThirdElement")
+		};
+
+		var hasDuplicates = dummyClasses.HasDuplicates(x => x.ComparableString);
+
+		Assert.True(hasDuplicates);
+	}
+
+	[Fact]
+	public async Task When_Asking_For_Duplicates_On_Sequence_With_Duplicates_Then_It_Does_Not_Iterate_Unnecessary_On_Elements()
```

Comment:
Given the complexity of testing and the duplication of testing (via the following test), this one is unnecessary.